portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,17 +38,50 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(json);
   }
 
+  @Override 
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Receive info from form
+    // String firstName = getParameter(request,"fname","");
+    // String lastName = getParameter(request,"lname","");
+    String information = getParameter(request,"info","");
+    boolean upperCase = Boolean.parseBoolean(getParameter(request,"upper-case","false"));
+    boolean sort = Boolean.parseBoolean(getParameter(request,"sort","false"));
+
+    // Apply Settings
+    if (upperCase) {
+      information = information.toUpperCase();
+    }
+
+    // Break up comma split list into words
+    String[] words = information.split("\\s*, \\s*");
+
```

Comment:
I may have forgotten my Java but do you need all the regex there? I believe .split(",") is sufficient.

https://www.geeksforgeeks.org/split-string-java-examples/

Reply:
I believe you can. Will change that next update